docs/en/04_Changelogs/4.0.0.md

```diff
@@ -1,197 +1,107 @@
-   for any list of candidates is actually performed by `SSViewer::chooseTemplate`
-
-### Front-end build tooling for CMS interface
-
- * Management of CMS JavaScript dependencies via [npm](https://www.npmjs.com/)
- * ES6 is now available in all core JavaScript files via [Babel](https://babeljs.io/)
- * Bundling of core JavaScript via [Browserify](http://browserify.org/)
- * CMS build tasks managed by [Gulp](http://gulpjs.com/)
-
-If you're planning to contribute to SilverStripe core, 
-you should learn how to [use the build tooling](/developer_guides/customising_the_admin_interface/cms_architecture)
-and [work with client-side dependencies](../contributing/code#working-with-client-side-dependencies). 
-Note that you don't need any of these tools for running SilverStripe or developing your own website. 
-
-### Added new front-end UI libraries
-
- * Added the following npm libraries:
-   * [react](https://www.npmjs.com/package/react)
-   * [redux](https://www.npmjs.com/package/redux)
-   * [react-router](https://github.com/reactjs/react-router)
-   * [react-router-redux](https://github.com/reactjs/react-router-redux)
-   * [react-redux](https://www.npmjs.com/package/react-redux)
-   * [react-dom](https://www.npmjs.com/package/react-dom)
-   * [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch)
-   * [redux-thunk](https://www.npmjs.com/package/redux-thunk)
-   * [react-addons-test-utils](https://www.npmjs.com/package/react-addons-test-utils)
- * Added SilverStripeComponent base class for building React UIs
-
-### Compiling CSS
-
-[Compass](http://compass-style.org/) has been removed and css/sprite compilation is now handled with [gulp](http://gulpjs.com/) (see [Markup and Style Conventions](../developer_guides/customising_the_admin_interface/cms_architecture/#markup-and-style-conventions)).
-
-### Bootstrap CSS
-
-Bootstrap 4 Alpha.2 CSS is available in `admin/css/bootstrap`.
-
-### Changed paths of static assets
+## Introduction
+
+This version introduces many breaking changes, which in most projects can be managed through a combination
+of automatic upgrade processes as well as manual code review. This document reviews these changes and will
+guide developers in preparing existing 3.x code for compatibility with 4.0
+
+* [Highlights of Major Changes](#overview)
+* [Upgrading Guide](#upgrading)
+  * [Standard Upgrade](#upgrading-primary)
+  * [API Specific Upgrades](#upgrading-specifics)
+* [API Changes](#api-changes)
+  * [General and Core API](#overview-general)
+  * [ORM API](#overview-orm)
+  * [Filesystem API](#overview-filesystem)
+  * [Template and Form API](#overview-template)
+* [Commit History](#commit-history)
+
+## <a name="overview"></a>Highlights of major changes
+
+* Minimum version dependencies have increased; PHP 5.5 and Internet Explorer 10 (or other modern browser)
+  is required.
+* All code earlier marked as deprecated for 4.0 has now been removed.
```

Comment:
Point to deprecation docs? If devs enable deprecation warnings they should be able to tell they're using deprecated features. Otherwise it's just causing concern here, since we don't list all removed APIs below